variables.tf

```diff
@@ -97,7 +97,8 @@ variable "applications" {
       redirect_uris = optional(set(string))
     }))
 
-    required_resource_access = optional(set(object({
+    required_resource_access = optional(map(object({
+      admin_consent   = optional(bool)
       resource_app_id = string
```

Comment:
We should expect the `object_id` of the Graph API Enterprise Application as input, from that we can look up the `application_id` using a `data` block.